Nitin Kushwaha/backend/eventHandlers.js

```diff
@@ -0,0 +1,25 @@
+const {addUser, users} = require('./users')
+
+module.exports = (io) => {
+    let user_no = 0;
+    function eventhandlers(socket){
+        console.log('Client connected!');
+
+        user_no++;
+        console.log(`${user_no} users connected`);
+
+        socket.on('user-name', (name)=>{
+            addUser({'id': socket.id, 'name': name});
+            io.emit('user-name', name);
+            io.emit('all-users', users);
+        })
+
+        socket.on('disconnect', ()=>{
+            console.log('Client disconnected!');
+
+            user_no--;
+            console.log(`${user_no} users connected`);
```

Comment:
```suggestion
            console.log('Client disconnected!');
            // Add a function to remove the user that just got disconnected from users array
            user_no--;
            console.log(`${user_no} users connected`);
```
You can do that, by adding a function in `users.js` file, something that would do:
```javascript
const index = users.findIndex((user) => user.id === id)
if (index !== -1) return users.splice(index, 1)
```